### PR TITLE
Add System Out as a logging source.

### PR DESCRIPTION
**Pull request by Souzooka**
Implements system out logging in PCSX2 console. The basic concept is explained in [this issue](https://github.com/PCSX2/pcsx2/issues/2558).

Example video:
https://gfycat.com/ThornyUnfinishedAplomadofalcon

In addition, this solution supports `printf` formatting, forgoing the need for the user to format the string in-game. By calling SYSCALL 0x75 with an unformatted string and arguments, PCSX2 will format the string.

This PR still needs a bit of polish. PCSX2 doesn't pull arguments from the stack to print them in this current implementation, meaning only up to 7 format arguments are used. If there is already a function that aggregates PS2 arguments it might be better utilized here as well. Additionally, another concern is that ConsoleLogSource always appends a newline character when writing text, and it's not necessarily something that may be preferable, though adding a method to write with ConsoleLogSource without a newline would be a lot of work and not necessarily within the scope of this PR. However, if you're pulling development/homebrew messages, these tend to already have newlines at the end of the message and thus you end up with having a lot of empty lines.

EDIT: To clarify on the newline thing, I think we can assume that if the user is a game developer and printing their own strings, or is a user that is restoring console functionality in a PS2 game that those strings already properly have a newline if necessary anyways. Thus I believe that the optimal solution probably would be a IConsoleWriter implementation that does not force newlines, and also provides formatting capabilities, though I would like to hear the opinions of others.

### CODE REVIEW

File: pcsx2/R5900OpcodeImpl.cpp [hunk 1]
```diff
@@ -983,7 +983,26 @@ void SYSCALL()
 			else
 				__Deci2Call(cpuRegs.GPR.n.a0.UL[0], (u32*)PSM(cpuRegs.GPR.n.a1.UL[0]));
 		}
-		break;
```
Comment (turtleli) [inline pcsx2/R5900OpcodeImpl.cpp]: I think `break;` shouldn't have been removed?

Comment (Souzooka) [reply pcsx2/R5900OpcodeImpl.cpp]: You're right, that's what I get for writing this late at night.

**Comment by ramapcsx2** [comment]
So I tested a few commercial games. None printed anything more than usual.
I was hoping it would :p 

Is this expected, or did I forget something?

**Comment by ramapcsx2** [comment]
Okay, this makes this basically a no op change for most people, but still useful to game modders and the like.
That's fine by me :)

Code looks fine as well, so if you want to leave it as is, I can merge it.

**Comment by Souzooka** [comment]
I did want to make it so messages don't have a newline, but maybe I'll add a ConsoleLogFromVM/ConsoleLogSource hybrid in a separate PR. If you want to merge this it should be a good first step and it stages this feature for a few tweaks later on.

**Comment by willkuer** [comment]
Actually what happens if this is called with more than 7 arguments? Will it crash, show the first seven arguments, show none of the arguments, not show the message at all?

Just curious.

**Comment by Souzooka** [comment]
IIRC, ConsoleLogSource will replace the arguments with defaults (e.g. "0" or "(null)"), like for example if you have the message
`"SET_POINT_CAMERA_NOW( %8.2f, %8.2f, %8.2f, %8.2f, %8.2f, %8.2f, %8.2f, %8.2f, %8.2f, %8.2f, %8.2f, %8.2f, 900, 0, POINT_CUBIC )"`, it might get printed out like `"SET_POINT_CAMERA_NOW( 9.675, 2, 10, 18, -20, 8.5, 2.125, 0, 0, 0, 0, 0, 900, 0, POINT_CUBIC)"`.

**Comment by ramapcsx2** [comment]
Just a quick question:
Is there an identifiable pattern to these no-op'd printf functions in games?
It'd be nice if PCSX2 could attempt to restore them automatically.

**Comment by Souzooka** [comment]
To get back to you on that, unfortunately I don't really know the answer to this since I'm really only interested in poking at Raw Danger, and other games may have different ways of doing debug, you would have to look at other games and see if there are certain patterns to stubbed printing functions.

**Pull request merged**